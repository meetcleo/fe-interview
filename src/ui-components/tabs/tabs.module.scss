@import '../../styles/variables.scss';

.tabsWrapper {
  display: flex;
  flex-direction: row;
  align-items: stretch;
  width: 100%;
  height: 48px;
  background: $blue;

  .tab {
    display: flex;
    justify-content: center;
    align-items: center;
    height: 100%;
    flex: 1;
    color: $white;
    text-decoration: none;
  }

  .tabActive {
    background: $yellow;
  }
}
