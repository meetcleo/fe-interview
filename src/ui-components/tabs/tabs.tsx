import React from 'react';
import { NavLink, RouteComponentProps, withRouter } from 'react-router-dom';

import styles from './tabs.module.scss';

export interface TabInfo {
  route: string;
  text: string;
}

export interface OwnProps {
  tabs: TabInfo[];
}

type Props = RouteComponentProps & OwnProps;

export const TabsComponent: React.FC<Props> = ({ tabs }) => (
  <div className={styles.tabsWrapper}>
    {tabs.map((tab) => {
      const { route, text } = tab;

      return (
        <NavLink
          key={text}
          className={styles.tab}
          activeClassName={styles.tabActive}
          to={route}
        >
          {text}
        </NavLink>
      );
    })}
  </div>
);

export const Tabs = withRouter(TabsComponent);
