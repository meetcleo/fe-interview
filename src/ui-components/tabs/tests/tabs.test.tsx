import React from 'react';
import { render } from 'enzyme';

import { MemoryRouter } from 'react-router-dom';
import { Tabs, OwnProps } from '../tabs';

describe('<Tabs >', () => {
  it('should render the component', () => {
    const mockProps: OwnProps = {
      tabs: [{ route: '/', text: 'Home' }],
    };
    const wrapper = render(
      <MemoryRouter>
        <Tabs {...mockProps} />
      </MemoryRouter>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
