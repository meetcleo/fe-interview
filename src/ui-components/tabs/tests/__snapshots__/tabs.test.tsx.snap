// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tabs > should render the component 1`] = `
<div
  class="tabsWrapper"
>
  <a
    aria-current="page"
    class="tab tabActive"
    href="/"
  >
    Home
  </a>
</div>
`;
